log/handler_glog.go

```diff
@@ -130,107 +126,84 @@ func (h *GlogHandler) Vmodule(ruleset string) error {
-// log whenever execution hits that statement.
-//
-// Unlike with Vmodule, the ".go" must be present.
-func (h *GlogHandler) BacktraceAt(location string) error {
-	// Ensure the backtrace location contains two non-empty elements
-	parts := strings.Split(location, ":")
-	if len(parts) != 2 {
-		return errTraceSyntax
-	}
-	parts[0] = strings.TrimSpace(parts[0])
-	parts[1] = strings.TrimSpace(parts[1])
-	if len(parts[0]) == 0 || len(parts[1]) == 0 {
-		return errTraceSyntax
-	}
-	// Ensure the .go prefix is present and the line is valid
-	if !strings.HasSuffix(parts[0], ".go") {
-		return errTraceSyntax
+func (h *GlogHandler) Enabled(ctx context.Context, lvl slog.Level) bool {
+	// fast-track skipping logging if override not enabled and the provided verbosity is above configured
+	return h.override.Load() || slog.Level(h.level.Load()) <= lvl
+}
+
+func (h *GlogHandler) WithAttrs(attrs []slog.Attr) slog.Handler {
+	h.lock.RLock()
+	siteCache := make(map[uintptr]slog.Level)
+	for k, v := range h.siteCache {
+		siteCache[k] = v
 	}
-	if _, err := strconv.Atoi(parts[1]); err != nil {
-		return errTraceSyntax
+	h.lock.RUnlock()
+
+	patterns := []pattern{}
+	patterns = append(patterns, h.patterns...)
+
+	res := GlogHandler{
+		origin:    h.origin.WithAttrs(attrs),
+		patterns:  patterns,
+		siteCache: siteCache,
+		location:  h.location,
 	}
-	// All seems valid
-	h.lock.Lock()
-	defer h.lock.Unlock()
 
-	h.location = location
-	h.backtrace.Store(len(location) > 0)
-	// Enable location storage (globally)
-	stackEnabled.Store(true)
-	return nil
+	res.level.Store(h.level.Load())
+	res.override.Store(h.override.Load())
+	return &res
+}
+
+func (h *GlogHandler) WithGroup(name string) slog.Handler {
+	panic("not implemented")
 }
 
 // Log implements Handler.Log, filtering a log record through the global, local
 // and backtrace filters, finally emitting it if either allow it through.
-func (h *GlogHandler) Log(r *Record) error {
-	// If backtracing is requested, check whether this is the callsite
-	if h.backtrace.Load() {
-		// Everything below here is slow. Although we could cache the call sites the
-		// same way as for vmodule, backtracing is so rare it's not worth the extra
-		// complexity.
-		h.lock.RLock()
-		match := h.location == r.Call.String()
-		h.lock.RUnlock()
-
-		if match {
-			// Callsite matched, raise the log level to info and gather the stacks
-			r.Lvl = LvlInfo
-
-			buf := make([]byte, 1024*1024)
-			buf = buf[:runtime.Stack(buf, true)]
-			r.Msg += "\n\n" + string(buf)
-		}
-	}
+func (h *GlogHandler) Handle(_ context.Context, r slog.Record) error {
 	// If the global log level allows, fast track logging
-	if h.level.Load() >= uint32(r.Lvl) {
-		return h.origin.Log(r)
-	}
-	// If no local overrides are present, fast track skipping
-	if !h.override.Load() {
-		return nil
+	if slog.Level(h.level.Load()) <= r.Level {
+		return h.origin.Handle(context.Background(), r)
 	}
+
```

Comment:
We should keep this, no? Seems like a valid optimization to me
 
```suggestion
	if !h.override.Load() {
		return nil
	}
```

Reply:
This is checked in `Enabled`.